### PR TITLE
Fix query result when skip=true (#1869)

### PR DESCRIPTION
**Pull request by edorivai**
Should fix https://github.com/apollographql/react-apollo/issues/1869

I added a test case, and took the implementation [from the original PR](https://github.com/apollographql/react-apollo/pull/1694/files#diff-45a733ae4e94396f8fa0947302418680R219) that aimed to implement the `skip` prop on `<Query>`.

I suspect this piece of logic got lost in the [big rewrite](https://github.com/apollographql/react-apollo/pull/1715) mentioned [here](https://github.com/apollographql/react-apollo/pull/1694#issuecomment-368563365).

### Checklist:

* [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
* [x] Make sure all of the significant new logic is covered by tests
* [ ] If this was a change that affects the external API used in GitHunt-React, update GitHunt-React and post a link to the PR in the discussion.

### CODE REVIEW

**Comment by ZwaarContrast** [comment]
Any progress on this?

**Comment by caseywebdev** [comment]
Ran into this issue yesterday, can a maintainer chime in here? I really like the `Query` component but I also need the `skip` and `loading` values to work correctly 😉

File: src/Query.tsx [hunk 1]
```diff
@@ -358,30 +358,46 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
...
-    if (loading) {
-      Object.assign(data.data, this.previousData, currentResult.data);
-    } else if (error) {
-      Object.assign(data, {
-        data: (this.queryObservable!.getLastResult() || {}).data,
-      });
+    // When skipping a query (ie. we're not querying for data but still want
+    // to render children), make sure the `data` is cleared out and
+    // `loading` is set to `false` (since we aren't loading anything).
+    if (this.props.skip) {
```
Comment (hwillson) [inline src/Query.tsx]: Quick note here - I've re-arranged this a bit to short circuit un-necessary extra query logic, when the query is to be skipped. One thing to add though, is that this means we're leaving `networkStatus` as `undefined`. This is currently intentional, as setting this to `NetworkStatus.ready` can cause problems when using something like the  `graphql` hoc and calculating the `skip` value using a function (forcing a `NetworkStatus.ready` value can prematurely end rendering). Leaving `networkStatus` as `undefined` here seems to be okay though, and is better than the way it was before when we did the `skip` check at the end of `getQueryResult`. That left `networkStatus` in `NetworkStatus.loading` state, even though we weren't loading anything.

**Comment by caseywebdev** [comment]
What's the status on this one? Seems like it's good to go (besides conflicts).

**Comment by rosskevin** [comment]
Please resolve conflicts, otherwise I think this good to merge

**Comment by edorivai** [comment]
@hwillson are you picking up the conflicts, or would you like me to do it?

**Comment by hwillson** [comment]
@edorivai I'll take care of it shortly. Thanks!

**Comment by stephane-ruhlmann** [comment]
Is anyone taking care of this?

**Comment by Kjagd** [comment]
@hwillson What's the status on this? We've been greatly affected by this for quite some time now.

**Comment by pascalstr** [comment]
@hwillson what is the status of resolving these conflicts and merging this PR?

**Comment by edorivai** [comment]
Great to hear :tada:

**Comment by sbogdaniuk** [comment]
SO you just fixed this issue for Component, and for HOC not, right?

**Pull request merged**